### PR TITLE
Tuist

### PR DESCRIPTION
**Pull request by fortmarek**
#### Checklist

- [x] Updated CHANGELOG.md.

[tuist](https://github.com/tuist/tuist) is a great tool that helps us with not committing `.xcodeproj` and it also makes the settings of the project more clearer as it is written in code.

Right now the template is more suited to projects that have one main project, but adding the ability to add additional ones (for modular architecture, frameworks, etc.) is planned and should be possible by modifying the current template a little bit.

The documentation for wiki is ready and I will update it once this is merged.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+target project_name do
+    project project_name + '.xcodeproj'
     pod 'SwiftLint', '~> 0.27'
     
-    pod 'ACKLocalization', '~> 0.3'
     pod 'SwiftGen', '~> 6.0'
     pod 'Smartling.i18n', '~> 1.0'
     
-    target 'UnitTests' do
+    target 'ProjectTemplateTests' do
```
Comment (olejnjak) [inline Podfile]: Is it really correct that the target here is named "ProjectTemplateTests"?

Comment (fortmarek) [reply Podfile]: This was done in an effort to be as easy as possible to add another project.
Adding a new project means using `project` definition from `Project+Templates` where the `tests` target is tied to the new project - which would mean a duplicated target if we change that to `UnitTests` and do not take the project's name into account.

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    private var configurationName: String {
+        switch self {
+        case .debug:
+            return "Debug"
+        case .betaDevelopment:
+            return "Beta-Development"
+        case .betaStage:
+            return "Beta-Stage"
+        case .betaProduction:
+            return "Beta-Production"
```
Comment (janmisar) [inline Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: Please revise also [Environments](https://github.com/AckeeCZ/iOS-MVVM-ProjectTemplate/wiki/Environments) section in wiki. These new configurations are not described there yet, I guess we forgot 🙈

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            "ENABLE_BITCODE": "NO",
+            "INFOPLIST_PREPROCESS": "YES",
+            "INFOPLIST_PREFIX_HEADER": "$(ACK_ENVIRONMENT_DIR)/.environment_preprocess.h",
+            "CODE_SIGN_STYLE": "Manual",
+        ]
+        
+        switch self {
+        case .debug:
+            let debugSettings: [String: SettingValue] = [
+                "ACK_BUNDLE_ID": SettingValue(stringLiteral: "cz.ackee.enterprise.\(name).\(identifierName)"),
```
Comment (janmisar) [inline Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: Do we still need `ACK_BUNDLE_ID`? It's same as `PRODUCT_BUNDLE_IDENTIFIER` and it has no additional value. Or am I wrong? 🤔

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        case .release:
+            return "Release"
+        }
+    }
+    
+    private func settings(with name: String, projectVersion: Version) -> [String: SettingValue] {
+        let base: [String: SettingValue] = [
+            "ACK_ENVIRONMENT_DIR": "$(PROJECT_DIR)/$(TARGET_NAME)/Environment",
+            "ACK_PROJECT_VERSION": SettingValue(stringLiteral: projectVersion.description),
+            "ACK_PRODUCT_NAME": SettingValue(stringLiteral: name),
```
Comment (janmisar) [inline Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: What is this for?

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift (hunk 3 above)
Comment (fortmarek) [reply Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: Yep, you're right, changed

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift (hunk 4 above)
Comment (fortmarek) [reply Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: This was due to the generalization I needed for a different project, but the way it is setup now it's not necessary, deleted, good catch 🙂

File: Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift (hunk 2 above)
Comment (fortmarek) [reply Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift]: it's ready, will push it once this is merged 🙂

**Pull request merged**